specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json

```diff
@@ -0,0 +1,1534 @@
+          "enum": [
+            "Basic",
+            "Standard"
+          ],
+          "x-ms-enum": {
+            "name": "AzureSkuTier",
+            "modelAsString": true
+          }
+        }
+      },
+      "description": "Azure SKU definition."
+    },
+    "CloudError": {
+      "x-ms-external": true,
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/CloudErrorBody",
+          "description": "An error response from Kusto."
+        }
+      },
+      "description": "An error response from Kusto."
+    },
+    "CloudErrorBody": {
+      "x-ms-external": true,
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "An identifier for the error. Codes are invariant and are intended to be consumed programmatically."
+        },
+        "message": {
+          "type": "string",
+          "description": "A message describing the error, intended to be suitable for displaying in a user interface."
+        },
+        "target": {
+          "type": "string",
+          "description": "The target of the particular error. For example, the name of the property in error."
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudErrorBody"
+          },
+          "description": "A list of additional details about the error."
+        }
+      },
+      "description": "An error response from Kusto."
+    },
+    "KustoPoolProperties": {
+      "properties": {
+        "state": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The state of the resource.",
+          "enum": [
+            "Creating",
+            "Unavailable",
+            "Running",
+            "Deleting",
+            "Deleted",
+            "Stopping",
+            "Stopped",
+            "Starting",
+            "Updating"
+          ],
+          "x-ms-enum": {
+            "name": "State",
+            "modelAsString": true
+          }
+        },
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The provisioned state of the resource."
+        },
+        "uri": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The Kusto Pool URI."
+        },
+        "dataIngestionUri": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The Kusto Pool data ingestion URI."
+        },
+        "stateReason": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The reason for the Kusto Pool's current state."
+        },
+        "engineType": {
+          "type": "string",
+          "description": "The engine type",
+          "enum": [
+            "V2",
+            "V3"
+          ],
+          "x-ms-enum": {
+            "name": "EngineType",
+            "modelAsString": true
+          }
+        },
+        "workspaceUid": {
```

Comment:
How is this workspace Id is different than workSpaceName parameter in the path? If they're refering to the same workspace, why would the service need both information?

Reply:
It's a requirement by Synapse that we (Kusto) are bound to as a sub-resource partner.
In a nutshell, as explained by the Synapse team regarding their sub-resource partner design, there are edge cases in which relying solely on using the workspaceName parameter in client requests could cause mismatch, while using workspace unique ID supports the decentralized architecture more properly.